src/manifolds/EssentialManifold.jl

```diff
@@ -0,0 +1,509 @@
+computing the distance of the equivalence classes $[p]$ and $[q]$ of the points
+$p=(R_{p_1},R_{p_2}), q=(R_{q_1},R_{q_2}) ∈ SO(3)^2$, respectively. Two points in $SO(3)^2$ are equivalent iff their
+corresponding essential matrices, given by
+````math
+E = R_1^T [e_z]_{×}R_2,
+````
+are equal (up to a sign flip). Using the logarithmic map, the distance is given by
+````math
+\text{dist}([p],[q]) = \| \text{log}_{[p]} [q] \| = \| \log_p (S_z(t_{\text{opt}})q) \|,
+````
+where $S_z ∈ H_z = \{(R_z(θ),R_z(θ))\colon θ \in [-π,π) \}$ in which $R_z(θ)$ is the rotation around the z axis with angle
+$θ$ and $t_{\text{opt}}$ is the minimizer of the cost function
+````math
+f(t) = f_1 + f_2, \quad f_i = \frac{1}{2} θ^2_i(t), \quad θ_i(t)=d(R_{p_i},R_z(t)R_{b_i}) \text{ for } i=1,2,
+````
+where $d(⋅,⋅)$ is the distance function in $SO(3)$[^TronDaniilidis2017].
+"""
+distance(M::EssentialManifold, p, q) = norm(M, p, log(M, p, q))
+
+@doc raw"""
+    exp(M::EssentialManifold, p, X)
+
+Compute the exponential map on the [`EssentialManifold`](@ref) from `p` into direction
+`X`, i.e.
+````math
+\text{exp}_p(X) =\text{exp}_g( \tilde X),  \quad g \in \text(SO)(3)^2,
+````
+where $\tilde X$ is the horizontal lift of $X$[^TronDaniilidis2017].
+"""
+exp(::EssentialManifold, ::Any...)
+
+function exp!(M::EssentialManifold, q, p, X)
+    exp!.(Ref(M.manifold), q, p, X)
+    return q
+end
+
+get_iterator(::EssentialManifold) = Base.OneTo(2)
+
+function isapprox(M::EssentialManifold, p, q; kwargs...)
+    return isapprox(distance(M, p, q), 0.0; kwargs...)
+end
+
+@doc raw"""
+    log(M::EssentialManifold, X, p, q)
+
+Compute the logarithmic map on the [`EssentialManifold`](@ref) `M`, i.e. the tangent vector,
+whose geodesic starting from `p` reaches `q` after time 1. Here, $p=(R_{p_1},R_{p_2})$ and
+$q=(R_{q_1},R_{q_2})$ are elements of $SO(3)^2$. We use that any essential matrix can, up to
+scale, be decomposed to
+````math
+E = R_1^T [e_z]_{×}R_2,
+````
+where $(R_1,R_2)∈SO(3)^2$. Two points in $SO(3)^2$ are equivalent iff their corresponding
+essential matrices are equal (up to a sign flip).
+To compute the logarithm, we first move `q` to another representative of its equivalence class.
+For this, we find $t= t_{\text{opt}}$ for which the function
+````math
+f(t) = f_1 + f_2, \quad f_i = \frac{1}{2} θ^2_i(t), \quad θ_i(t)=d(R_{p_i},R_z(t)R_{b_i}) \text{ for } i=1,2,
+````
+where $d(⋅,⋅)$ is the distance function in $SO(3)$, is minimized. Further, the group $H_z$ acting
+on the left on $SO(3)^2$ is defined as
+````math
+H_z = \{(R_z(θ),R_z(θ))\colon θ \in [-π,π) \},
+````
+where $R_z(θ)$ is the rotation around the z axis with angle $θ$. Points in $H_z$ are denoted by
+$S_z$. Then, the logarithm is defined
+as
+````math
+\log_p (S_z(t_{\text{opt}})q) = [\text{Log}(R_{p_i}^T R_z(t_{\text{opt}})R_{b_i})]_{i=1,2},
+````
+where $\text{Log}$ is the [`logarithm`](@ref log(::Rotations, ::Any...)) on $SO(3)$. For more
+details see [^TronDaniilidis2017].
+"""
```

Comment:
This docstring should be attached to `log`, not `log!`.

Reply:
Ah, yes, thanks for noticing.